File: js/euler004.js
/*
  A palindromic number reads the same both ways. The largest palindrome made
  from the product of two 2-digit numbers is 9009 = 91 × 99.

  Find the largest palindrome made from the product of two 3-digit numbers.
 */

var helper = require('./helper.js'),

    one = 999,
    two = 999,
    level,
    result = one * two;

function isPal(num) {
  num = num.toString();

  return num === num
    .split('')
    .reverse()
    .join('');
}

while (!isPal(result)) {
  level = 1;
  while (!isPal(result) && level < 10) {
    result = one * (two - level++);
  }
  one--;
}

helper(result);

File: lua/euler004.lua
--[[
  A palindromic number reads the same both ways. The largest palindrome made
  from the product of two 2-digit numbers is 9009 = 91 × 99.

  Find the largest palindrome made from the product of two 3-digit numbers.
  ]]

local helper = require("lua/helper").helper
local level

local one = 999
local two = 999

local product = one * two

function isPal (num)
  return string.lower(num) == string.reverse(string.lower(num))
end

while not isPal(product) do
  level = 1
  while not isPal(product) and level < 10 do
    level = level + 1
    product = one * (two - level)
  end
  one = one - 1
end

helper(906609, product)
